### PR TITLE
Es6 refactor module kind plugin

### PR DESCRIPTION
**Pull request by rachel-ftw**

### CODE REVIEW

File: lib/ModuleKindPlugin.js [hunk 1]
```diff
@@ -1,26 +1,29 @@
+"use strict";
```
Comment (sokra) [inline lib/ModuleKindPlugin.js]: use strict should be after the license comment

File: lib/ModuleKindPlugin.js [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
+		this.target = target;
+	}
+	apply(resolver) {
+		const target = this.target;
+		resolver.plugin(this.source, (request, callback) => {
+			if(!request.module) return callback();
+			const obj = assign({}, request);
+			delete obj.module;
+			resolver.doResolve(target, obj, "resolve as module", createInnerCallback((err, result) => {
+				if(arguments.length > 0) return callback(err, result);
```
Comment (sokra) [inline lib/ModuleKindPlugin.js]: `arguments` is not available in an arrow function

File: lib/ModuleKindPlugin.js [hunk 3]
```diff
@@ -1,26 +1,29 @@
+"use strict";
 /*
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var createInnerCallback = require("./createInnerCallback");
+const assign = require("object-assign");
```
Comment (sokra) [inline lib/ModuleKindPlugin.js]: Use Object.assign

**Comment by ooflorent** [comment]
Thank you for your PR but it appears that the file was previously converted to ES6. I'm closing the PR for this reason.

**Pull request closed without merging**